### ISSUE TITLE
Getting Assertion `wd[len - 1] == L'/'' failed over SSH

### ISSUE THREAD

**Issue by davispuh** [fish-shell/fish-shell#2298]
Locally everything works fine, but when I ssh into and start writing "c" (that's only letter which triggers it) then I get this assert
```
fish: src/path.cpp:148: bool path_get_cdpath(const wcstring&, wcstring*, const wchar_t*, const env_vars_snapshot_t&): Assertion `wd[len - 1] == L'/'' failed.
```

```
#0  0x00007f2a148eb5f8 in raise () from /usr/lib/libc.so.6
#1  0x00007f2a148eca7a in abort () from /usr/lib/libc.so.6
#2  0x00007f2a148e4417 in __assert_fail_base () from /usr/lib/libc.so.6
#3  0x00007f2a148e44c2 in __assert_fail () from /usr/lib/libc.so.6
#4  0x00000000004acb64 in path_get_cdpath (dir=L"AUR", out=out@entry=0x7f2a1439eda0,
    wd=0x70aaf8 <std::basic_string<wchar_t, std::char_traits<wchar_t>, std::allocator<wchar_t> >::_Rep::_S_empty_rep_storage@@GLIBCXX_3.4+24> L"", env_vars=...)
    at src/path.cpp:148
#5  0x000000000045e074 in autosuggest_validate_from_history (item=..., detector=..., working_directory=L"", vars=...) at src/highlight.cpp:532
#6  0x0000000000476b68 in threaded_autosuggest (this=0x836df0) at src/reader.cpp:1468
#7  threaded_autosuggest (ctx=0x836df0) at src/reader.cpp:1522
#8  0x00000000004af7c0 in iothread_worker (unused=<optimized out>) at src/iothread.cpp:135
#9  0x00007f2a151f74a4 in start_thread () from /usr/lib/libpthread.so.0
#10 0x00007f2a149a112d in clone () from /usr/lib/libc.so.6
```

Using latest git just compiled, but it also happens with previous versions.

**Comment by davispuh** [comment]
Actually it also happens when I login from TTY, but only for that one user and not in Konsole under X. It might be because I've symlinked $HOME

**Comment by zanchey** [comment]
I think this should be fixed with #2553 - could you check the latest build?

**Comment by davispuh** [comment]
yeah, no more assert :) now it's just

```
fish: Could not determine current working directory. Is your locale set correctly?
Welcome to fish, the friendly interactive shell
Type help for instructions on how to use fish
fish: Tried to print invalid wide character string

fish: Tried to print invalid wide character string
```
